File: fast_sync_dotnet_service/fast-sync-core/abstraction/metadata/ISyncMetadata.cs
namespace fast_sync_core.abstraction.data
{
    public interface ISyncMetadata: IWithId
    {
        string SyncZone { get; set; }
        string Type { get; set; }
        int Version { get; set; }
        long Timestamp { get; set; }
        SyncOperationEnum SyncOperation { get; set; }
        string GetSyncZone();
        string ComputeSyncZone(SyncZoneRestrictionEnum syncZoneRestrictionType);
    }
}

File: fast-sync-typescript-client-sdk/packages/fast-sync-client/src/abstraction/metadata/isync_metadata.ts
import { SyncZoneRestrictionEnum } from "../models/sync_zone_restriction";
import { SyncOperationEnum } from "./isync_operation";
import { IWithId } from "./iwith_id";

export interface ISyncMetadata extends IWithId {
  id: string;
  syncZone: string;
  type: string;
  version: number;
  timestamp: number;
  syncOperation: SyncOperationEnum;
  getSyncZone();
  computeSyncZone(syncZoneRestrictionType: SyncZoneRestrictionEnum);
}
